src/state/pools/index.ts

```diff
@@ -298,6 +305,17 @@ export const fetchCakeVaultUserData = createAsyncThunk<SerializedLockedVaultUser
   },
 )
 
+export const fetchIfoUserCreditDataAsync =
+  (account: string): AppThunk =>
+  async (dispatch) => {
+    try {
+      const iCakeUserData = await fetchIfoUserCredit(account)
+      dispatch(setIfoUserCreditData(iCakeUserData))
```

Comment:
Imo No need to dispatch again, we can do this operation inside the slice directly

Reply:
@memoyil 
may I ask why you suggest we don't need to dispatch here, cound't find the difference between the pattern like "fetchPoolsUserDataAsync"?
thanks for your kindly help🙏

Reply:
@memoyil 
may I ask why you suggest we don't need to dispatch here, cound't find the difference between the pattern like "fetchPoolsUserDataAsync"?
thanks for your kindly help🙏

Reply:
Hey @memoyil , changed the state. 
But not sure what you mean in here.Or do we have any example?

Reply:
You can create a reducer in extraReducers of poolsSlice for fetchIfoUserCreditDataAsync.fulfilled then change the state inside that reducer. Therefore we will not need to call and start the dispatch process again

Reply:
@memoyil  Thanks.